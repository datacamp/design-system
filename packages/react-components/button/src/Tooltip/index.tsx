import Card from '@datacamp/waffles-card';
import Positioner from '@datacamp/waffles-positioner';
import { Text } from '@datacamp/waffles-text';
import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { css } from '@emotion/core';
import React, { MutableRefObject, ReactElement } from 'react';

interface TooltipProps {
  appearance: 'light' | 'dark';
  children: string;
  id: string;
  position:
    | 'bottom'
    | 'top'
    | 'right'
    | 'left'
    | 'bottomLeft'
    | 'bottomRight'
    | 'topLeft'
    | 'topRight';
  target: MutableRefObject<HTMLElement | undefined>;
  visible: boolean;
}

const marginMap = {
  bottom: css({ marginTop: tokens.size.space[8].value }),
  bottomLeft: css({ marginTop: tokens.size.space[8].value }),
  bottomRight: css({ marginTop: tokens.size.space[8].value }),
  left: css({ marginRight: tokens.size.space[8].value }),
  right: css({ marginLeft: tokens.size.space[8].value }),
  top: css({ marginBottom: tokens.size.space[8].value }),
  topLeft: css({ marginBottom: tokens.size.space[8].value }),
  topRight: css({ marginBottom: tokens.size.space[8].value }),
};

const cardStyles = {
  dark: css({
    backgroundColor: tokens.color.primary.navy.value.hex,
    boxShadow: 'none', // override default card border
  }),
  light: css({
    backgroundColor: tokens.color.neutral.beige100.value.hex,
  }),
};

const textStyles = {
  dark: css({ color: tokens.color.primary.white.value.hex }),
  light: css({ color: tokens.color.primary.navyText.value.hex }),
};

const elevations = { dark: 0, light: 2 } as const;

const Tooltip = ({
  appearance,
  children,
  id,
  visible,
  target,
  position,
}: TooltipProps): ReactElement | null => {
  return (
    <Positioner position={position} target={target} visible={visible}>
      <Card
        css={css(
          {
            paddingBottom: tokens.size.space[8].value,
            paddingLeft: tokens.size.space[12].value,
            paddingRight: tokens.size.space[12].value,
            paddingTop: tokens.size.space[8].value,
          },
          marginMap[position],
          cardStyles[appearance],
        )}
        elevation={elevations[appearance]}
        id={id}
      >
        <Text
          css={css(
            {
              fontSize: tokens.size.font[200].value,
              whiteSpace: 'nowrap',
            },
            textStyles[appearance],
          )}
        >
          {children}
        </Text>
      </Card>
    </Positioner>
  );
};

export default Tooltip;
