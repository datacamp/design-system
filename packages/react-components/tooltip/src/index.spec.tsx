import axeRender from '@datacamp/waffles-axe-render';
import React from 'react';

import Tooltip from './index';

describe('Tooltip', () => {
  ([
    'bottomLeft',
    'bottom',
    'bottomRight',
    'left',
    'right',
    'topLeft',
    'top',
    'topRight',
  ] as const).forEach((position) => {
    (['light', 'dark'] as const).forEach((appearance) => {
      it(`renders on the ${position} on the target`, async () => {
        const ref = React.createRef<HTMLDivElement>();
        const { baseElement } = await axeRender(
          <>
            <div
              css={{ height: 50, width: 50 }}
              data-test-id="target"
              ref={ref}
            />
            <Tooltip
              appearance={appearance}
              id="abc"
              position={position}
              target={ref}
              visible
            >
              tooltip text content
            </Tooltip>
          </>,
        );

        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
