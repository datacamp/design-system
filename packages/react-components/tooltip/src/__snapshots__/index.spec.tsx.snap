// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders on the bottom on the target 1`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25),0 2px 4px -1px rgba(5,25,45,0.3);
  position: fixed;
  left: 0;
  top: 8px;
  -webkit-transform: translate(-50%, 0);
  -moz-transform: translate(-50%, 0);
  -ms-transform: translate(-50%, 0);
  transform: translate(-50%, 0);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #fffbf3;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #05192d;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the bottom on the target 2`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25);
  position: fixed;
  left: 0;
  top: 8px;
  -webkit-transform: translate(-50%, 0);
  -moz-transform: translate(-50%, 0);
  -ms-transform: translate(-50%, 0);
  transform: translate(-50%, 0);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #05192d;
  box-shadow: none;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #ffffff;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the bottomLeft on the target 1`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25),0 2px 4px -1px rgba(5,25,45,0.3);
  position: fixed;
  left: 0;
  top: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #fffbf3;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #05192d;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the bottomLeft on the target 2`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25);
  position: fixed;
  left: 0;
  top: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #05192d;
  box-shadow: none;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #ffffff;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the bottomRight on the target 1`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25),0 2px 4px -1px rgba(5,25,45,0.3);
  position: fixed;
  left: 0;
  top: 8px;
  -webkit-transform: translate(-100%);
  -moz-transform: translate(-100%);
  -ms-transform: translate(-100%);
  transform: translate(-100%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #fffbf3;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #05192d;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the bottomRight on the target 2`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25);
  position: fixed;
  left: 0;
  top: 8px;
  -webkit-transform: translate(-100%);
  -moz-transform: translate(-100%);
  -ms-transform: translate(-100%);
  transform: translate(-100%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #05192d;
  box-shadow: none;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #ffffff;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the left on the target 1`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25),0 2px 4px -1px rgba(5,25,45,0.3);
  position: fixed;
  left: -8px;
  top: 0;
  -webkit-transform: translate(-100%, -50%);
  -moz-transform: translate(-100%, -50%);
  -ms-transform: translate(-100%, -50%);
  transform: translate(-100%, -50%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #fffbf3;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #05192d;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the left on the target 2`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25);
  position: fixed;
  left: -8px;
  top: 0;
  -webkit-transform: translate(-100%, -50%);
  -moz-transform: translate(-100%, -50%);
  -ms-transform: translate(-100%, -50%);
  transform: translate(-100%, -50%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #05192d;
  box-shadow: none;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #ffffff;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the right on the target 1`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25),0 2px 4px -1px rgba(5,25,45,0.3);
  position: fixed;
  left: 8px;
  top: 0;
  -webkit-transform: translate(0, -50%);
  -moz-transform: translate(0, -50%);
  -ms-transform: translate(0, -50%);
  transform: translate(0, -50%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #fffbf3;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #05192d;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the right on the target 2`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25);
  position: fixed;
  left: 8px;
  top: 0;
  -webkit-transform: translate(0, -50%);
  -moz-transform: translate(0, -50%);
  -ms-transform: translate(0, -50%);
  transform: translate(0, -50%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #05192d;
  box-shadow: none;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #ffffff;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the top on the target 1`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25),0 2px 4px -1px rgba(5,25,45,0.3);
  position: fixed;
  left: 0;
  top: -8px;
  -webkit-transform: translate(-50%, -100%);
  -moz-transform: translate(-50%, -100%);
  -ms-transform: translate(-50%, -100%);
  transform: translate(-50%, -100%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #fffbf3;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #05192d;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the top on the target 2`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25);
  position: fixed;
  left: 0;
  top: -8px;
  -webkit-transform: translate(-50%, -100%);
  -moz-transform: translate(-50%, -100%);
  -ms-transform: translate(-50%, -100%);
  transform: translate(-50%, -100%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #05192d;
  box-shadow: none;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #ffffff;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the topLeft on the target 1`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25),0 2px 4px -1px rgba(5,25,45,0.3);
  position: fixed;
  left: 0;
  top: -8px;
  -webkit-transform: translate(0, -100%);
  -moz-transform: translate(0, -100%);
  -ms-transform: translate(0, -100%);
  transform: translate(0, -100%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #fffbf3;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #05192d;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the topLeft on the target 2`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25);
  position: fixed;
  left: 0;
  top: -8px;
  -webkit-transform: translate(0, -100%);
  -moz-transform: translate(0, -100%);
  -ms-transform: translate(0, -100%);
  transform: translate(0, -100%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #05192d;
  box-shadow: none;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #ffffff;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the topRight on the target 1`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25),0 2px 4px -1px rgba(5,25,45,0.3);
  position: fixed;
  left: 0;
  top: -8px;
  -webkit-transform: translate(-100%, -100%);
  -moz-transform: translate(-100%, -100%);
  -ms-transform: translate(-100%, -100%);
  transform: translate(-100%, -100%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #fffbf3;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #05192d;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;

exports[`Tooltip renders on the topRight on the target 2`] = `
.emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  -webkit-transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: all 600ms cubic-bezier(0.075, 0.82, 0.165, 1);
  box-shadow: 0 0 2px 0 rgba(5, 25, 45, 0.25);
  position: fixed;
  left: 0;
  top: -8px;
  -webkit-transform: translate(-100%, -100%);
  -moz-transform: translate(-100%, -100%);
  -ms-transform: translate(-100%, -100%);
  transform: translate(-100%, -100%);
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  z-index: 1;
  background-color: #05192d;
  box-shadow: none;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: #05192d;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-style: normal;
  font-size: 14px;
  font-weight: 400;
  font-size: 12px;
  white-space: nowrap;
  color: #ffffff;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-test-id="target"
    />
    <div
      class="emotion-1"
      id="abc"
    >
      <span
        class="emotion-2"
      >
        tooltip text content
      </span>
    </div>
  </div>
</body>
`;
